src/main/java/com/mentorship/vineservice/controllers/VineController.java

```diff
@@ -0,0 +1,53 @@
+package com.mentorship.vineservice.controllers;
+
+
+import com.mentorship.vineservice.dto.IdDto;
+import com.mentorship.vineservice.domain.Vine;
+import com.mentorship.vineservice.dto.PaginationDto;
+import com.mentorship.vineservice.services.imp.VineServiceImpl;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequiredArgsConstructor
+public class VineController {
+
+    private static final String TOTAL_COUNT_HEADER = "X-Total-Count";
+    private final VineServiceImpl vineServiceImpl;
+
+    @PostMapping("/vine/create")
```

Comment:
It is better to have just the same url signature (f.e /vine) with different HTTP methods